src/test/java/com/connexta/search/index/IndexServiceImplTest.java

```diff
@@ -6,239 +6,175 @@
-                    .index(datasetId, irmUri));
-    assertThat(thrown.getStatus(), is(HttpStatus.BAD_REQUEST));
-    assertThat(thrown.getCause(), is(ioException));
+            SearchException.class, () -> indexService.index(Index.builder().id(datasetId).build()));
+    assertThat(thrown.getCause(), is(runtimeException));
   }
 
-  @ParameterizedTest
-  @MethodSource("exceptionsThrownSearchManager")
-  void testSearchManagerExceptions(final RuntimeException runtimeException) throws Exception {
+  @Test
+  void testExceptionWhenSaving() throws Exception {
+    // given
     final String datasetId = "00067360b70e4acfab561fe593ad3f7a";
+    final String contents =
+        "All the color had been leached from Winterfell until only grey and white remained.";
 
-    final RuntimeException thrown;
-
-    try (final MockWebServer storeMockWebServer = new MockWebServer()) {
-      storeMockWebServer.start();
-      storeMockWebServer.enqueue(
-          new MockResponse()
-              .setBody(
-                  String.format(
-                      "{ \"%s\" : \"All the color had been leached from Winterfell until only grey and white remained.\" }",
-                      EXT_EXTRACTED_TEXT))
-              .setResponseCode(200));
-      final URI irmUri = storeMockWebServer.url(String.format("/dataset/%s/irm", datasetId)).uri();
-      doThrow(runtimeException)
-          .when(mockSearchManager)
-          .index(eq(datasetId), eq(irmUri), any(InputStream.class));
-
-      thrown =
-          assertThrows(
-              RuntimeException.class,
-              () ->
-                  ((IndexService) new IndexServiceImpl(mockSearchManager, WebClient.create()))
-                      .index(datasetId, irmUri));
-    }
-
-    assertThat(
-        "thrown exception is the exact same exception thrown by the SearchManager",
-        thrown,
-        is(runtimeException));
+    final URI irmUri = new URI(String.format("http://store:9041/dataset/%s/irm", datasetId));
+    final URI fileUri = new URI(String.format("http://store:9041/dataset/%s/", datasetId));
+    final Index index =
+        Index.builder()
+            .id(datasetId)
+            .contents(contents)
+            .irmUrl(irmUri.toString())
+            .fileUrl(fileUri.toString())
+            .build();
+    doReturn(IOUtils.toInputStream("asdf", StandardCharsets.UTF_8))
+        .when(mockGetInputStream)
+        .get(anyString());
+    doReturn("fileContents").when(mockBodyContentExtractor).extractText(any(InputStream.class));
+
+    // and stub CrudRepository#existsById
+    when(mockIndexRepository.existsById(datasetId)).thenReturn(false);
+
+    // and stub CrudRepository#save
+    final RuntimeException runtimeException = new RuntimeException();
+    doThrow(runtimeException)
+        .when(mockIndexRepository)
+        .save(
+            argThat(
+                anIndex -> {
+                  return anIndex.equals(index);
+                }));
```

Comment:
✏️ 
```java
.save(eq(index));
```

You also don't need to explicitly check the argument here because that's already tested in `testIndex`.
```java
.save(any());
```
is good enough.